cores/xmc/WInterrupts.c

```diff
@@ -0,0 +1,200 @@
+//****************************************************************************
+// @Project Includes
+//****************************************************************************
+#include "Arduino.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+//****************************************************************************
+// @Prototypes Of Local Functions
+//****************************************************************************
+static voidFuncPtr  interrupt_0_cb = NULL;
+static voidFuncPtr  interrupt_1_cb = NULL;
+
+//****************************************************************************
+// @Local Functions
+//****************************************************************************
+#if defined(INTERRUPT_USE_ERU)
+void ERU0_3_IRQHandler(void) {
+    if (interrupt_0_cb) {
+        interrupt_0_cb();
+    }
+}
+
+    #if defined(XMC4200_Platform2GO)
+void ERU0_0_IRQHandler(void) {
+    if (interrupt_1_cb) {
+        interrupt_1_cb();
+    }
+}
+    #else
+void ERU1_0_IRQHandler(void) {
+    if (interrupt_1_cb) {
+        interrupt_1_cb();
+    }
+}
+    #endif
+
+#else
+void CCU40_0_IRQHandler(void) {
+    if (interrupt_0_cb) {
+        interrupt_0_cb();
+    }
+}
+
+void CCU40_1_IRQHandler(void) {
+    if (interrupt_1_cb) {
+        interrupt_1_cb();
+    }
+}
+#endif
+
+void attachInterrupt(pin_size_t  interrupt_num, voidFuncPtr  callback, PinStatus  mode) {
+        XMC_PIN_INTERRUPT_t pin_irq = mapping_interrupt[interrupt_num];
+#if defined(INTERRUPT_USE_ERU)
+        XMC_ERU_ETL_EDGE_DETECTION_t event_config = 0;
+
+        switch (mode) {
+        case CHANGE:
+            event_config = XMC_ERU_ETL_EDGE_DETECTION_BOTH;
+            break;
+
+        case RISING:
+            event_config = XMC_ERU_ETL_EDGE_DETECTION_RISING;
+            break;
+
+        case FALLING:
+            event_config = XMC_ERU_ETL_EDGE_DETECTION_FALLING;
+            break;
+
+        default:
+            event_config = XMC_CCU4_SLICE_EVENT_EDGE_SENSITIVITY_NONE;
+            break;
+        }
+
+        XMC_ERU_Enable(pin_irq.eru);
+        XMC_ERU_ETL_SetInput(pin_irq.eru, pin_irq.etl, pin_irq.input_a, pin_irq.input_b);
+        XMC_ERU_ETL_SetEdgeDetection(pin_irq.eru, pin_irq.etl, event_config);
+        XMC_ERU_ETL_SetSource(pin_irq.eru, pin_irq.etl, XMC_ERU_ETL_SOURCE_B);
+        XMC_ERU_ETL_EnableOutputTrigger(pin_irq.eru, pin_irq.etl, pin_irq.ogu);
+        XMC_ERU_OGU_SetServiceRequestMode(pin_irq.eru, pin_irq.ogu,
+                                          XMC_ERU_OGU_SERVICE_REQUEST_ON_TRIGGER);
+
+        if (pin_irq.irq_num == 0) {
+            NVIC_SetPriority(ERU0_3_IRQn, 3);
+            interrupt_0_cb = callback;
+            NVIC_EnableIRQ(ERU0_3_IRQn);
+        } else if (pin_irq.irq_num == 1) {
+    #if defined(XMC4200_Platform2GO)
```

Comment:
Should these be handled by defines in the pins_arduino.h ?
That way we can avoid here all these cascaded if/else that need to be modified for each board.